src/sampleutil.cpp

```diff
@@ -296,26 +296,31 @@ void SampleUtil::convertS16ToFloat32(CSAMPLE* _RESTRICT pDest, const SAMPLE* _RE
 }
 
 // static
-bool SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
+SampleUtil::CLIPPING SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
         const CSAMPLE* pBuffer, int iNumSamples) {
     CSAMPLE fAbsL = CSAMPLE_ZERO;
     CSAMPLE fAbsR = CSAMPLE_ZERO;
-    CSAMPLE clipped = 0;
+    CSAMPLE clippedL = 0;
+    CSAMPLE clippedR = 0;
```

Comment:
Oh! I see.
I needed to put a static_cast there though. It would be like this:

if (absr > CSAMPLE_PEAK) {
    clipped = static_cast<SampleUtil::CLIP_STATUS>(clipped | SampleUtil::CLIPPING_RIGHT);
}

Does the if affect the loop vectorizing?